### ISSUE TITLE
Enhance the screenshot to support Chrome (and other browsers)

### ISSUE THREAD

**Issue by zpsean** [jonsamwell/flutter_gherkin#210]
As stated in https://github.com/jonsamwell/flutter_gherkin/issues/191. Currently the flutter_gherkin has issue to define the step for screenshot on Chrome (and other browsers). 

[**Note**: for web, flutter web can not take screenshot by itself, it will notify the flutter_driver, then flutter_driver notify the chromedriver to tell Chrome to take screenshot with API call, after done, the flutter_driver then notify integration_test for further test steps]

Per the https://github.com/flutter/flutter/tree/master/packages/integration_test, the sample to take screenshot is using the binding:

`await binding.takeScreenshot('screenshot-1');`

```
void main() {
  final binding = IntegrationTestWidgetsFlutterBinding.ensureInitialized()
      as IntegrationTestWidgetsFlutterBinding;

  testWidgets('screenshot', (WidgetTester tester) async {
    // Build the app.
    app.main();

    // Trigger a frame.
    await tester.pumpAndSettle();
    await binding.takeScreenshot('screenshot-1');
  });
}
```

In flutter_gherkin，it is found that the _binding is claimed in gherkin_integration_test_runner.dart ：
https://github.com/jonsamwell/flutter_gherkin/blob/1913832893576e391cf48b0953b62079177190da/lib/src/flutter/runners/gherkin_integration_test_runner.dart#L53-L64

and hooks

https://github.com/jonsamwell/flutter_gherkin/blob/1913832893576e391cf48b0953b62079177190da/lib/src/flutter/hooks/attach_screenshot_on_failed_step_hook.dart#L25-L29

may need to be extended with logic, in Pseudocode: 

```
Future<String> takeScreenshot(World world) async { 
    if ( kIsWeb) {
        final bytes = await _binding.takeScreenshot('screenshot-1');
    }
   else {
       final bytes = await (world as FlutterWorld).appDriver.screenshot();
   }
    return base64Encode(bytes);
  }
```

Now the issue is how to pass the claimed _binding to the step definition?

**Comment by jonsamwell** [comment]
Thanks @zpsean the binding variable is probably something that should be given the the driver adapter class.  I'll try and look into the next week.  Thanks for the pseudocode

**Comment by jonsamwell** [comment]
We should probably pass it to the adapter here 

https://github.com/jonsamwell/flutter_gherkin/blob/1913832893576e391cf48b0953b62079177190da/lib/src/flutter/runners/gherkin_integration_test_runner.dart#L248

Then for the above screenshot logic in the adapter rather than the step definition / hook

**Comment by zpsean** [comment]
I'm considering for screenshot, we may have below 3 scenarios:
- Implicitly: take screenshot for all Failed steps [currently in hooks]
- Implicitly: take screenshot for all steps
- Explicitly: User may want to define his own steps to take screenshot as required (like. When I take screenshot 'screenshot-homepage-01')

**Comment by jonsamwell** [comment]
Yes this covers most scenarios I think

**Comment by jonsamwell** [comment]
Should be fixed in the latest release `3.0.0-rc.10`
